hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRequest.java

```diff
@@ -132,6 +132,12 @@ protected KeyArgs resolveBucketLink(
       int preallocateBlocksMax, boolean grpcBlockTokenEnabled, String omID)
       throws IOException {
 
+    // avoid contact with scm for OMThroughputBenchmark test
+    if (preallocateBlocksMax < 1) {
+      LOG.debug("preallocateBlocksMax smaller than 1, skip allocateBlock");
+      return new ArrayList<>();
+    }
```

Comment:
Can we avoid production code change for benchmark?

Reply:
This is designed to avoid contacting SCM to alleviate the extra influence. Here is also a logic bug, if the config ```ozone.key.preallocation.max.blocks``` is 0(preallocateBlocksMax =0), OM will still contact SCM to allocateBlock, which is unreasonable. Do you think it shall be fixed?